homeassistant/components/freedompro/climate.py

```diff
@@ -0,0 +1,144 @@
+"""Support for Freedompro climate."""
+import json
+import logging
+
+from pyfreedompro import put_state
+
+from homeassistant.components.climate import ClimateEntity
+from homeassistant.components.climate.const import (
+    ATTR_HVAC_MODE,
+    HVAC_MODE_COOL,
+    HVAC_MODE_HEAT,
+    HVAC_MODE_OFF,
+    SUPPORT_TARGET_TEMPERATURE,
+)
+from homeassistant.const import ATTR_TEMPERATURE, CONF_API_KEY, TEMP_CELSIUS
+from homeassistant.core import callback
+from homeassistant.helpers import aiohttp_client
+from homeassistant.helpers.update_coordinator import CoordinatorEntity
+
+from .const import DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+HVAC_MAP = {
+    0: HVAC_MODE_OFF,
+    1: HVAC_MODE_HEAT,
+    2: HVAC_MODE_COOL,
+}
+
+HVAC_INVERT_MAP = {
+    HVAC_MODE_OFF: 0,
+    HVAC_MODE_HEAT: 1,
+    HVAC_MODE_COOL: 2,
+}
+
+SUPPORTED_HVAC_MODES = [HVAC_MODE_OFF, HVAC_MODE_HEAT, HVAC_MODE_COOL]
+
+
+async def async_setup_entry(hass, entry, async_add_entities):
+    """Set up Freedompro climate."""
+    api_key = entry.data[CONF_API_KEY]
+    coordinator = hass.data[DOMAIN][entry.entry_id]
+    async_add_entities(
+        Device(hass, api_key, device, coordinator)
+        for device in coordinator.data
+        if device["type"] == "thermostat"
+    )
+
+
+class Device(CoordinatorEntity, ClimateEntity):
+    """Representation of an Freedompro climate."""
+
+    def __init__(self, hass, api_key, device, coordinator):
+        """Initialize the Freedompro climate."""
+        super().__init__(coordinator)
+        self._session = aiohttp_client.async_get_clientsession(hass)
+        self._api_key = api_key
+        self._attr_name = device["name"]
+        self._attr_unique_id = device["uid"]
+        self._characteristics = device["characteristics"]
+        self._attr_device_info = {
+            "name": self.name,
+            "identifiers": {
+                (DOMAIN, self.unique_id),
+            },
+            "model": device["type"],
+            "manufacturer": "Freedompro",
+        }
+        self._attr_supported_features = SUPPORT_TARGET_TEMPERATURE
+        self._attr_current_temperature = 0
+        self._attr_target_temperature = 0
+        self._attr_temperature_unit = TEMP_CELSIUS
+        self._attr_hvac_mode = HVAC_MODE_OFF
+        self._attr_hvac_modes = SUPPORTED_HVAC_MODES
```

Comment:
The constant attributes can be defined as class attributes as they will never change.

Reply:
Sorry I didn't understand what you mean

Reply:
Define the constant attributes as class attributes instead of instance attributes:
```py
class SomeEntity(Entity):
    _attr_hvac_modes = SUPPORTED_HVAC_MODES
```

Reply:
Note that generally mutable objects should not be defined as class attributes to avoid bugs, but for constants it's since they should only be defined in once place and never updated.